components/brave_rewards/resources/android_page/components/pending_contributions_modal.tsx

```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+
+import Profile from '../../ui/components/profile/index'
+import { DetailRow } from '../../ui/components/tablePending/index'
+import { LocaleContext, formatMessage } from '../../shared/lib/locale_context'
+import { Modal, ModalCloseButton } from '../../shared/components/modal'
+import Tokens from '../../ui/components/tokens/index'
+import { TrashIcon } from './icons/trash_icon'
+
+import * as style from './pending_contributions_modal.style'
+
+interface Props {
+  rows: DetailRow[]
+  onClose: () => void
+}
+
+export function PendingContributionsModal(props: Props) {
+  const { getString } = React.useContext(LocaleContext)
+  function getRow(row: DetailRow) {
+    /* Each row is a grid
+       ---------------------------------------------
+       |   | 1(29px) | 2(1fr) |  3(1fr)  | 4(14px) |
+       ---------------------------------------------
+       | 1 |  Icon     Site              |     |_| |
+       | 2 |         | Pending Amount              |
+       | 3 |         | Type  |                Date |
+       ---------------------------------------------
+    */
+     return (
+      <style.constributionsListItem>
+        <style.contributionReceiver>
+          <a href={row.url} target={'_blank'}>
+            <Profile
+              title={row.profile.name}
+              provider={row.profile.provider}
+              verified={row.profile.verified}
+              src={row.profile.src}
+              type='mobile'
+            />
+          </a>
+        </style.contributionReceiver>
+        <style.contributionDelete>
+          <button onClick={row.onRemove}><TrashIcon /></button>
+        </style.contributionDelete>
+        <style.contributionAmount>
+          <Tokens
+            value={row.amount.tokens}
+            converted={row.amount.converted}
+            size={'small'}
+            showApproxSign={true}
+          />
+        </style.contributionAmount>
+        <style.contributionType>
+          {getString(`pendingType${row.type}`)}
```

Comment:
Yes! I meant to do that, but then it ran away from me. Added this function into `tablePending` component, so that we can use it in both places - in [11e3046](https://github.com/brave/brave-core/pull/12565/commits/11e3046b3f20c933f5f24a6f4be991499920c89b)